arwen/host/execution.go

```diff
@@ -246,18 +258,31 @@ func (host *vmHost) ExecuteOnDestContext(input *vmcommon.ContractCallInput) (vmO
 
 func (host *vmHost) handleBuiltinFunctionCall(input *vmcommon.ContractCallInput) (*vmcommon.ContractCallInput, *vmcommon.VMOutput, error) {
 	output := host.Output()
+
+	asyncPrefixArgs := arwen.PopCallIDsFromArguments(input)
+	if asyncPrefixArgs == nil {
+		// this should never happen
+		err := fmt.Errorf("Async framework error - PopCallIDsFromArguments")
```

Comment:
define and error for this. and return that error directly - delete the // this should never happen comment

Reply:
done